src/blot/abstract/shadow.ts

```diff
@@ -40,8 +43,7 @@ class ShadowBlot implements Blot {
 
   public prev: Blot | null;
   public next: Blot | null;
-  // @ts-expect-error
-  public parent: Parent;
+  public parent!: Parent;
```

Comment:
Let's keep `// @ts-expect-error` as `!` is not an actual fix. `parent` indeed can be absent before `attach` is called.